From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Sat, 19 Oct 2013 12:14:12 +0300
Subject: [PATCH 106/223] RDMA/cxgb4: Fix underflows in c4iw_create_qp()
Patch-mainline: v3.15-rc1
Git-commit: ff1706f4feb8e0e1a2e56a8dd57e17a4b45649b5
References: bsc#909577 FATE#317550

These sizes should be unsigned so we don't allow negative values and
have underflow bugs.  These can come from the user so there may be
security implications, but I have not tested this.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: David Chang <dchang@suse.com>
Signed-off-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/cxgb4/qp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/infiniband/hw/cxgb4/qp.c
+++ b/drivers/infiniband/hw/cxgb4/qp.c
@@ -1535,7 +1535,7 @@ struct ib_qp *c4iw_create_qp(struct ib_p
 	struct c4iw_cq *schp;
 	struct c4iw_cq *rchp;
 	struct c4iw_create_qp_resp uresp;
-	int sqsize, rqsize;
+	unsigned int sqsize, rqsize;
 	struct c4iw_ucontext *ucontext;
 	int ret;
 	struct c4iw_mm_entry *mm1, *mm2, *mm3, *mm4, *mm5 = NULL;
